clicache/src/ReflectionBasedAutoSerializer.cpp

```diff
@@ -228,18 +254,30 @@ namespace Apache
                 break;
 			          case FieldType::BYTE:
                   return r->ReadByte(m_fieldName);
+                break;
+			          case FieldType::SBYTE:
+                  return r->ReadByte(m_fieldName);
```

Comment:
This didn't require any explicit casting like the others?

Reply:
No. Remember ReadByte returns an SByte already. A very bogus API name for a .NET developer. This is one of the main reasons we need to clean up the existing API.